### PR TITLE
Add tests for `ConvolutionFFT`

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        assertEquals(expected.size(), result.size(), "Signal lengths are not equal.");
+        for (int i = 0; i < expected.size(); i++) {
+            FFT.Complex expectedValue = expected.get(i);
+            FFT.Complex resultValue = result.get(i);
+            assertEquals(expectedValue.real(), resultValue.real(), delta, "Real part mismatch at index " + i);
+            assertEquals(expectedValue.imaginary(), resultValue.imaginary(), delta, "Imaginary part mismatch at index " + i);
+        }
+    }
+
+    @Test
```
Comment (alxkm) [inline src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java:34]: Will be better to use `@ParameterizedTest` for this and additional class to store data.

File: src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public void testConvolutionFFTBasic() {
+        double[] a = {1, 2, 3};
```
Comment (alxkm) [inline src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java:36]: This can be put directly to avoid a lot of code duplication

**Review by alxkm — CHANGES_REQUESTED** [review]

File: src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java (hunk 1 above)
Comment (Hardvan) [reply src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java]: Done!

File: src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java (hunk 2 above)
Comment (Hardvan) [reply src/test/java/com/thealgorithms/maths/ConvolutionFFTTest.java]: Done!

**Comment by Hardvan** [comment]
I have resolved the conflicts @alxkm .

**Review by alxkm — APPROVED** [review]

**Pull request merged**